dask_ml/model_selection/_incremental.py

```diff
@@ -0,0 +1,349 @@
+from __future__ import division
+
+import operator
+from copy import deepcopy
+from time import time
+
+import dask
+import dask.array as da
+from dask.distributed import Future, default_client, futures_of, wait
+from distributed.utils import log_errors
+from sklearn.base import clone
+from sklearn.utils import check_random_state
+from tornado import gen
+
+
+def _partial_fit(model_and_meta, X, y, fit_params):
+    """
+    Call partial_fit on a classifiers with training data X and y
+
+    Arguments
+    ---------
+    model_and_meta : Tuple[Estimator, dict]
+    X, y : np.ndarray, np.ndarray
+        Training data
+    fit_params : dict
+        Extra keyword arguments to pass to partial_fit
+
+    Returns
+    -------
+    model : Estimator
+        The model that has been fit.
+    meta : dict
+        A new dictionary with updated information.
+    """
+    with log_errors():
+        start = time()
+        model, meta = model_and_meta
+
+        if len(X):
+            model = deepcopy(model)
+            model.partial_fit(X, y, **(fit_params or {}))
+
+        meta = dict(meta)
+        meta["partial_fit_calls"] += 1
+        meta["partial_fit_time"] = time() - start
+
+        return model, meta
+
+
+def _score(model_and_meta, X, y, scorer):
+    start = time()
+    model, meta = model_and_meta
+    if scorer:
+        score = scorer(model, X, y)
+    else:
+        score = model.score(X, y)
+
+    meta = dict(meta)
+    meta.update(score=score, score_time=time() - start)
+    return meta
+
+
+def _create_model(model, ident, **params):
+    """ Create a model by cloning and then setting params """
+    with log_errors(pdb=True):
+        model = clone(model).set_params(**params)
+        return model, {"model_id": ident, "params": params, "partial_fit_calls": 0}
+
+
+@gen.coroutine
+def _fit(
+    model,
+    params,
+    X_train,
+    y_train,
+    X_test,
+    y_test,
+    additional_calls,
+    fit_params=None,
+    scorer=None,
+    random_state=None,
+):
+    original_model = model
+    fit_params = fit_params or {}
+    client = default_client()
+    rng = check_random_state(random_state)
+
+    info = {}
+    models = {}
+    scores = {}
+
+    for ident, param in enumerate(params):
+        model = client.submit(_create_model, original_model, ident, **param)
+        info[ident] = []
+        models[ident] = model
+
+    # assume everything in fit_params is small and make it concrete
+    fit_params = yield client.compute(fit_params)
+
+    # Convert testing data into a single element on the cluster
+    # This assumes that it fits into memory on a single worker
+    if isinstance(X_test, da.Array):
+        X_test = client.compute(X_test)
+    else:
+        y_test = yield client.scatter(y_test)
```

Comment:
Should this be `X_test = yield client.scatter(X_test)`?

Reply:
Yes, it probably should.  My mistake.